crates/types/src/consensus.rs

```diff
@@ -263,9 +263,57 @@ impl Default for ConsensusMetricsValue {
 }
 
 impl<TYPES: NodeType> Consensus<TYPES> {
+    /// Constructor.
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        validated_state_map: BTreeMap<TYPES::Time, View<TYPES>>,
+        cur_view: TYPES::Time,
+        last_decided_view: TYPES::Time,
+        saved_leaves: CommitmentMap<Leaf<TYPES>>,
+        saved_payloads: BTreeMap<TYPES::Time, Arc<[u8]>>,
+        locked_view: TYPES::Time,
+        high_qc: QuorumCertificate<TYPES>,
+        metrics: Arc<ConsensusMetricsValue>,
+    ) -> Self {
+        Consensus {
+            validated_state_map,
+            vid_shares: BTreeMap::new(),
+            saved_da_certs: HashMap::new(),
+            cur_view,
+            last_decided_view,
+            saved_leaves,
+            saved_payloads,
+            locked_view,
+            high_qc,
+            metrics,
+            dontuse_decided_upgrade_cert: None,
+            dontuse_formed_upgrade_certificate: None,
+        }
+    }
+
+    /// Get the current view.
+    pub fn cur_view(&self) -> TYPES::Time {
+        self.cur_view
+    }
+
     /// Update the current view.
-    pub fn update_view(&mut self, view_number: TYPES::Time) {
-        self.cur_view = view_number;
+    pub fn update_view_if_new(&mut self, view_number: TYPES::Time) {
+        if view_number > self.cur_view {
+            self.cur_view = view_number;
+        }
+    }
+
+    /// Get the high QC.
+    pub fn high_qc(&self) -> &QuorumCertificate<TYPES> {
```

Comment:
Can this be `get_high_qc` to be more consistent with our naming?

Reply:
I used to debate this and found both `get_something` and `something` in our codebase. It's not consistent anyway, and since I slightly prefer the latter, I didn't add ` get`.